mona_sdk/client.py

```diff
@@ -632,7 +664,80 @@ def get_config_history(self, number_of_revisions=UNPROVIDED_VALUE):
             "get_config_history", data={"number_of_revisions": number_of_revisions}
         )
 
+    def _update_sampling_factors_if_needed(self):
```

Comment:
I think it'll be much better to cache this function, rather than get_sampling_factors.
1. it will save redundant work done in this function if the results are cached
2. the get_sampling_factors is external and might be invoked in other ways that won't want it to be cached...